web_src/js/features/notification.js

```diff
@@ -27,35 +45,59 @@ export function initNotificationCount() {
 
   if (NotificationSettings.EventSourceUpdateTime > 0 && !!window.EventSource) {
     // Try to connect to the event source first
-    const source = new EventSource(`${AppSubUrl}/user/events`);
-    source.addEventListener('notification-count', async (e) => {
-      try {
-        const data = JSON.parse(e.data);
-
-        const notificationCount = $('.notification_count');
-        if (data.Count === 0) {
-          notificationCount.addClass('hidden');
-        } else {
-          notificationCount.removeClass('hidden');
-        }
 
-        notificationCount.text(`${data.Count}`);
-        await updateNotificationTable();
-      } catch (error) {
-        console.error(error);
-      }
-    });
-    source.addEventListener('logout', async (e) => {
-      if (e.data !== 'here') {
-        return;
-      }
-      source.close();
-      window.location.href = AppSubUrl;
-    });
-    window.addEventListener('beforeunload', () => {
-      source.close();
-    });
-    return;
+    if (window.SharedWorker) {
+      const worker = new SharedWorker(`${__webpack_public_path__}js/eventsource.sharedworker.js`, 'notification-worker');
+      worker.addEventListener('error', (event) => {
+        console.error(event);
+      }, false);
+      worker.port.onmessageerror = () => {
+        console.error('Unable to deserialize message');
+      };
+      worker.port.postMessage({
+        type: 'start',
+        url: `${window.location.protocol}//${window.location.host}${AppSubUrl}/user/events`,
+      });
+      worker.port.addEventListener('message', (e) => {
+        if (!e.data || !e.data.type) {
+          console.error(e);
+          return;
+        }
+        switch (event.data.type) {
+          case 'notification-count':
+            receiveUpdateCount(e.data);
+            return;
+          case 'error':
+            console.error(e.data);
+            return;
+          case 'logout': {
+            if (e.data !== 'here') {
+              return;
+            }
+            worker.port.postMessage({
+              type: 'close',
+            });
+            worker.port.close();
+            window.location.href = AppSubUrl;
+            return;
+          }
+          default:
+            return;
+        }
+      }, false);
+      worker.port.addEventListener('error', (e) => {
+        console.error(e);
+      }, false);
+      worker.port.start();
+      window.addEventListener('beforeunload', () => {
+        worker.port.postMessage({
+          type: 'close',
+        });
+        worker.port.close();
+      }, false);
```

Comment:
```suggestion
      });
```

Reply:
The MDN documentation explicitly recommends false on addEventListener.

Reply:
That's outdated info. Any modern browser defaults it to `false` now, see `useCapture parameter made optional` in the compat table on mdn.